### PR TITLE
adds doc describing settings DSL

### PR DESCRIPTION
**Pull request by stockholmux**
**Issue number:**

Contributes to #203 

**Description of changes:**

This adds meta-documentation on a DSL to describe in semi-structured TOML

**Terms of contribution:**

By submitting this pull request, I confirm that my contribution is made under
the terms of the licenses outlined in the LICENSE-SUMMARY file.

### CODE REVIEW

**Review by stmcginnis — COMMENTED** [review]
Great start. A few cleanup items pointed out.

I am having a hard time following the plan, or at least forming a mental model in my mind about what this is describing. I know when I've heard you talk about it before it made a lot of sense. So maybe just a little more background context at the beginning describing the need and how it addressed maybe?

Or else it's just Friday afternoon and my mind is a little too scrambled. Which could very well be. ;)

It does start to make sense, so I think especially once it's rendered with the screenshots, it will be a little easier to digest. And I do think it is mostly clear what someone would need to do to document settings, so that's great. And once we have some real world docs in place, this will be a great supplement to help understand it.

Anyway, great to have this written down, and despite the comments above I think it will be good to have captured and will really help to get settings information captured and documented this way.

File: METADOC-SETTING-DOCS-DSL.md [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+* Tags (“topics”) can convey information but only one-to-many information that contextually applies to all settings within the same tag. 
+* Use the structured key/value fields as much as possible and only use free-form options when the structured keys do not apply. However, any documentation is better than none!
+
+---
+
+Documentation for settings are expressed as[TOML v1.0.0](https://toml.io/en/v1.0.0).
+
+All documentation starts at the key `docs` . 
+
+Settings in Bottlerocket are represented by at least a 2-level nesting, and most commonly 3-level nesting starting with `settings` . A single instance of `docs` represents the settings under a particular 2nd level nesting For example, documenting `settings.foo` would be accomplished by one instance of `docs` and `settings.bar` by another.
```
[the remark this answers is not in the document]
Comment (stockholmux) [reply METADOC-SETTING-DOCS-DSL.md:23]: great catch on the spacing. The perils of exporting from another tools into markdown.

WRT last sentence, I think the metasyntactic variables are getting in the way of understanding this. I'll revise.

**Comment by stockholmux** [comment]
@stmcginnis I added a high-level overview to address the mental model gap.

**Comment by stockholmux** [comment]
See `ecs.toml` in #206 for a more extensive example.

**Pull request merged**